flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/tasks/StreamConfigChainer.java

```diff
@@ -42,15 +42,17 @@
 /**
  * Helper class to build StreamConfig for chain of operators.
  */
-public class StreamConfigChainer {
+public class StreamConfigChainer<OWNER> {
+	private final OWNER owner;
```

Comment:
I didn't get why we'd need the owner in this commit.

Reply:
It's in order to provide a nice builder like API, for nested builder. That after `finish()` call on setting up the chaining, we  can continue configuring or build the test harness:
```
			new MultipleInputStreamTaskTestHarnessBuilder<>(MultipleInputStreamTask::new, BasicTypeInfo.STRING_TYPE_INFO)
				.addInput(BasicTypeInfo.STRING_TYPE_INFO)
				.addInput(BasicTypeInfo.STRING_TYPE_INFO)
				.addInput(BasicTypeInfo.STRING_TYPE_INFO)
				.setupOperatorChain(new DuplicatingOperator())
				.chain(new OneInputStreamTaskTest.DuplicatingOperator(), BasicTypeInfo.STRING_TYPE_INFO.createSerializer(new ExecutionConfig()))
				.chain(new OneInputStreamTaskTest.DuplicatingOperator(), BasicTypeInfo.STRING_TYPE_INFO.createSerializer(new ExecutionConfig()))
				.finish()
				.setTaskMetricGroup(taskMetricGroup)
				.build()
```
And `OWNER` is template, so that `finish()` and hence `build()` would return correct specific type of the test harness (`MultipleInput`)

Reply:
Gotcha. StreamConfigChainer is only used as a builder anyway, right?

Reply:
Yes, exactly. It is a builder to setup operator chain.